src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -95,47 +105,77 @@ public void visitToken(DetailAST ast) {
                 break;
 
             case TokenTypes.CLASS_DEF:
-                registerNestedSubclassToOuterSuperClasses(ast);
-
-                final boolean isFinal = modifiers.branchContains(TokenTypes.FINAL);
-                final boolean isAbstract = modifiers.branchContains(TokenTypes.ABSTRACT);
-
-                final String qualifiedClassName = getQualifiedClassName(ast);
-                classes.push(new ClassDesc(qualifiedClassName, isFinal, isAbstract));
+                currentDepthLevel++;
+                final ClassDesc classDesc = buildClassDesc(ast, modifiers);
+                classes.push(classDesc);
```

Comment:
Why a `buildClassDesc` and not a `registerClassDesc` like you did for `CTOR_DEF`?